### PR TITLE
[klein] create a new port 

### PR DESCRIPTION
**Pull request by JPMMaia**
**Describe the pull request**

- #### What does your PR fix?  
  It adds a new port for [klein](https://github.com/jeremyong/klein).

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  all, Yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  - Updated the version files
  - Formatted manifests

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by JPMMaia** [comment]
> -- Configuring x86-windows
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:105 (message):
    Command failed: ninja -v
    Working Directory: D:/buildtrees/klein/x86-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      D:\buildtrees\klein\config-x86-windows-out.log

> Call Stack (most recent call first):
  D:/installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:351 (vcpkg_execute_required_process)
  ports/klein/portfile.cmake:9 (vcpkg_cmake_configure)
  scripts/ports.cmake:139 (include)
  
  It seems that the build system can't execute `ninja -v`? Should I add `DISABLE_PARALLEL_CONFIGURE` to `vcpkg_cmake_configure`?

**Comment by JonLiu1993** [comment]
> > -- Configuring x86-windows
> > CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:105 (message):
> > Command failed: ninja -v
> > Working Directory: D:/buildtrees/klein/x86-windows-rel/vcpkg-parallel-configure
> > Error code: 1
> > See logs for more information:
> > D:\buildtrees\klein\config-x86-windows-out.log
> 
> > Call Stack (most recent call first):
> > D:/installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:351 (vcpkg_execute_required_process)
> > ports/klein/portfile.cmake:9 (vcpkg_cmake_configure)
> > scripts/ports.cmake:139 (include)
> 
> It seems that the build system can't execute `ninja -v`? Should I add `DISABLE_PARALLEL_CONFIGURE` to `vcpkg_cmake_configure`?

@JPMMaia ,Thanks for your pr,
From the error log on CI, you don’t need to add DISABLE_PARALLEL_CONFIGURE to vcpkg_cmake_configure. this is the latest error log of CI. Could you please take a look?

[config-x86-windows-out.log](https://github.com/microsoft/vcpkg/files/6606712/config-x86-windows-out.log)

File: ports/klein/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        -DKLEIN_ENABLE_TESTS=OFF
+        -DKLEIN_VALIDATE=OFF
+        -DKLEIN_BUILD_SYM=OFF
+        -DKLEIN_BUILD_C_BINDINGS=OFF
+        -DKLEIN_INSTALL=ON
+)
+
+vcpkg_cmake_install()
+
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/klein RENAME copyright)
```
Comment (JonLiu1993) [inline ports/klein/portfile.cmake:22]: Please change'.../share/klein' to '.../share/${PORT}'

File: ports/klein/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+file(INSTALL ${CMAKE_CURRENT_LIST_DIR}/usage DESTINATION ${CURRENT_PACKAGES_DIR}/share/klein)
```
Comment (JonLiu1993) [inline ports/klein/portfile.cmake:23]: Please change'.../share/klein' to '.../share/${PORT}'

File: ports/klein/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        -DKLEIN_ENABLE_PERF=OFF
+        -DKLEIN_ENABLE_TESTS=OFF
+        -DKLEIN_VALIDATE=OFF
+        -DKLEIN_BUILD_SYM=OFF
+        -DKLEIN_BUILD_C_BINDINGS=OFF
+        -DKLEIN_INSTALL=ON
+)
+
+vcpkg_cmake_install()
+
```
Comment (JonLiu1993) [inline ports/klein/portfile.cmake:21]: Please add 'vcpkg_fixup_pkgconfig()'

File: ports/klein/vcpkg.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "klein",
+  "version-string": "2.3.0",
```
Comment (JonLiu1993) [inline ports/klein/vcpkg.json:3]: The latest [version ](https://github.com/jeremyong/klein/releases)of klein is 2.2.1,could you explain why it is 2.3.0 here

We should use "version" instead "version-string"

File: versions/k-/klein.json [hunk 5]
```diff
@@ -0,0 +1,9 @@
+{
+  "versions": [
+    {
+      "git-tree": "49c209248e2837e7eccefd0e6d94fe3f9e6beedd",
+      "version-string": "2.3.0",
```
Comment (JonLiu1993) [inline versions/k-/klein.json:5]: We should use "version" instead"version-string"

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/klein/vcpkg.json (hunk 4 above)
Comment (JPMMaia) [reply ports/klein/vcpkg.json]: It's currently using [this commit](https://github.com/jeremyong/klein/commit/9c1879601d9c7c5a8f82b88cf853f5426fe3e81f).
These are changes that are needed for vcpkg to work (without adding more patches). These changes are not present in the latest release (2.2.1). Should we use a version date instead?

Comment (JonLiu1993) [reply ports/klein/vcpkg.json]: Yes, we need to change it to a date. Only when the latest stable version is released can we use the version representation

File: ports/klein/vcpkg.json [hunk 6]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "klein",
+  "version": "2.3.0",
```
Comment (JonLiu1993) [inline ports/klein/vcpkg.json:3]: Similarly, if you change the date, you also need to modify “version” to "version-date"

File: versions/k-/klein.json [hunk 7]
```diff
@@ -0,0 +1,9 @@
+{
+  "versions": [
+    {
+      "git-tree": "49c209248e2837e7eccefd0e6d94fe3f9e6beedd",
+      "version": "2.3.0",
```
Comment (JonLiu1993) [inline versions/k-/klein.json:5]: Similarly, if you change the date, you also need to modify “version” to "version-date"

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by JonLiu1993** [comment]
@JPMMaia ,If this pr is ready to be reviewed, please let me know

**Comment by JPMMaia** [comment]
@JonLiu1993 thanks for updating the git version and fixing the builds! I marked it as ready for review.

**Review by JonLiu1993 — APPROVED** [review]

File: ports/klein/find_simde_first.patch [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
++        simde
++        GIT_REPOSITORY https://github.com/simd-everywhere/simde
++        GIT_TAG v0.7.2
++        GIT_SHALLOW ON
++    )
++    FetchContent_GetProperties(simde)
++    if(NOT simde_POPULATED)
++        FetchContent_Populate(simde)
++    endif()
+ endif()
```
Comment (strega-nil-ms) [inline ports/klein/find_simde_first.patch:34]:
```suggestion
-if(NOT simde_POPULATED)
-    FetchContent_Populate(simde)
-endif()
+find_path(SIMDE_INCLUDE_DIR "simde/x86/sse4.1.h" REQUIRED)
+set(simde_SOURCE_DIR ${SIMDE_INCLUDE_DIR}/simde)
```

File: ports/klein/find_simde_first.patch [hunk 9]
```diff
@@ -0,0 +1,36 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index c5d6f3f..a3ba161 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -29,16 +29,21 @@ endif()
```
Comment (strega-nil-ms) [inline ports/klein/find_simde_first.patch:5]:
```suggestion
@@ -29,16 +29,7 @@ endif()
```

**Review by strega-nil-ms — CHANGES_REQUESTED** [review]
we never want to download simde, we should just fail.

**Comment by JonLiu1993** [comment]
@JPMMaia ,Please run the command "./vcpkg x-add-version klein --overwrite-version" then submit the change again

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JPMMaia** [comment]
@strega-nil-ms I believe I applied the changes you requested.

**Review by strega-nil-ms — APPROVED** [review]
Thanks @JPMMaia :)

**Pull request merged**